src/SFML/Window/Unix/ClipboardImpl.cpp

```diff
@@ -0,0 +1,268 @@
+#include <SFML/Window/Unix/Display.hpp>
+#include <SFML/System/String.hpp>
+#include <SFML/System/Thread.hpp>
+#include <SFML/System/Mutex.hpp>
+#include <SFML/System/Sleep.hpp>
+#include <iostream>
+#include <string>
+#include <X11/Xlib.h>
+
+
+namespace
+{
+////////////////////////////////////////////////////////////
+    void initClipboard();
+    void* hostSelection(void*);
+
+    sf::String string;
+
+    pthread_mutex_t mutex;
+    pthread_t host_thread;
+    
+    bool is_fail = false;
+    bool is_init = false;
+    bool is_host = false;
+
+    Display* display = NULL;
+    Window window = 0;
+
+    Atom selection = 0;
+    Atom atom_targ = 0;
+    Atom atom_text = 0;
+    Atom utf8_text = 0;
+    int xa_string = 31;
+    int xa_atom = 4;
+
+////////////////////////////////////////////////////////////
+    void initClipboard()
+    {
+        is_init = true;
+
+        display = XOpenDisplay(NULL);
+        int screen = DefaultScreen(display);
+        window = XCreateSimpleWindow(display, RootWindow(display, screen), 
+            0, 0, 1, 1, 0, BlackPixel(display, screen), WhitePixel(display, screen));
+
+        selection = XInternAtom(display, "CLIPBOARD", false);
+        atom_targ = XInternAtom(display, "TARGETS", false);
+        atom_text = XInternAtom(display, "TEXT", false);
+        utf8_text = XInternAtom(display, "UTF8_STRING", true);
+
+        if(utf8_text == None)
+        {
+            std::cerr << "UTF8 format unavailable on clipboard." << std::endl;
+            utf8_text = xa_string;
+        }
+
+        if(pthread_mutex_init(&mutex, NULL))
+        {
+            is_fail = true;
+            std::cerr << "Unable to initialize mutex. Failed to initialize Clipboard." << std::endl;
+            return;
+        }
+
+        if(pthread_create(&host_thread, NULL, hostSelection, NULL))
+        {
+            is_fail = true;
+            std::cerr << "Unable to create host thread. Failed to initialize Clipboard." << std::endl;
+            return;
+        }
+    }
+
+////////////////////////////////////////////////////////////
+    void* hostSelection(void*)
+    {
+        while(true)
+        {
+            if(XPending(display) && is_host)
+            {
+                XEvent event;
+
+                pthread_mutex_lock(&mutex);
+                XNextEvent(display, &event);
+                pthread_mutex_unlock(&mutex);
+
+                switch(event.type)
+                {
+                    case SelectionClear:
+                    {
+                        pthread_mutex_lock(&mutex);
+                        is_host = false;
+                        pthread_mutex_unlock(&mutex);
+
+                        break;
+                    }
+                    case SelectionRequest:
+                    {
+                        if(event.xselectionrequest.selection == selection)
+                        {
+                            std::cerr << "Sending clipboard data" << std::endl;
```

Comment:
Oops yeah, there is still some cleaning up I need to do. Thanks for spotting that.